### ISSUE TITLE
Confusing Documentation: Relation to TermInterface.jl

### ISSUE THREAD

**Issue by samysweb** [JuliaSymbolics/SymbolicUtils.jl#505]
Hello,
after about 4 hours of debugging a version update of SymbolicUtils.jl today, I stumbled across [this commit](https://github.com/JuliaSymbolics/SymbolicUtils.jl/commit/50c266ce4c707eed5a35780a092bb8e57751fde8).
It is my understanding that SymbolicUtils.jl is now working independently from TermInterface.jl and one should therefore no longer implement `istree`,`operation`,`arguments`, etc. for `TermInterface`, but for `SymbolicUtils` directly.
This is an important distinction, because my custom interfacing with SymbolicUtils so far looked as below which does not work with the new version of SymbolicUtils that has the TermInterface integrated into its own code base.
```julia
import TermInterface.istree
# ...

istree(x::MyType) = true
# ...
```
It would be nice to eventually have an update for the [documentation on interfacing](https://symbolicutils.juliasymbolics.org/interface/) concerning this change since it currently still tells you to interface with `TermInterface` which will not work and is a terrible thing to debug by hand.
I'm mainly opening this issue so that other people with similar problems may find it until the documentation has been updated accordingly.

Nonetheless, thank you very much for your great work on this project, it has really helped me a lot in my project!

**Comment by ReubenJ** [comment]
The package is relying on and implementing `TermInterface` again as of 2.0 (https://github.com/JuliaSymbolics/SymbolicUtils.jl/commit/4b3ae15c6a8c1c8a007f4842ea90b5cf845bbe6f). Looks like this is no longer relevant, right?

**Comment by oameye** [comment]
@ReubenJ Yes exactly. But now you only have to implement `TermInterface.head` and optionally `TermInterface.metadata`. See for example:
https://github.com/qojulia/QuantumCumulants.jl/blob/95499faa09dcbb8000d4d35529de5cb953e3b3ab/src/qnumber.jl#L44-L48

**Comment by oameye** [comment]
I still think the documentation could be improved on how to properly define an interface. Maybe we should add an example.

**Comment by ReubenJ** [comment]
> But now you only have to implement `TermInterface.head` and optionally `TermInterface.metadata`.

Is that generally true? It seems surprising to me that you don't also need `TermInterface.children`, but I'm not yet familiar with `SymbolicUtils` or `TermInterface`.

For interface definition, an `Interfaces.jl`-style definition could be a nice-to-have. That way, users of the package can easily test whether they're implementing the interface correctly—no need to rely only on reading the documentation.
